### DISCUSSION TITLE
Create Deta Base DB on app start (FastApi)

### DISCUSSION THREAD

**Discussion by Master-Y0da** [deta/cloud-docs#218] [Help] [answered]
I'm trying to create a users DB in Deta base when app is starting, but can't make it work.

I try using event startup:

      @app.on_event("startup")
        def startup_event():
       users = deta.Base("users")
       user = users.put({
        "name": "name",
        "age": "age",
        "hometown": "hometown"
      })

And also tryed with __main__ :

         if __name__ == '__main__':
          users = deta.Base("users")
          user = users.put({
        "name": "name",
        "age": "age",
        "hometown": "hometown"
       })

Any help would be so much appreciated!.

**Comment by abdelhai** [comment]
what is the error you are getting? how are you making sure it doesn't work?

**Reply by Master-Y0da** [reply]
There is no error...just nothing happens when I take a look into my app url...
Expected behaviour would be that I enter into my Deta panel, and in base section should display an item called **users**

When I deploy code to deta, behind scenes is restarting/re launching the app ?

**Reply by abdelhai** [reply]
i recommend following this tutorial exactly: https://docs.deta.sh/docs/base/py_tutorial

events like `on_startup` don't work on deta micros

**Reply by Master-Y0da** [reply]
Ok !...but what about if I need to populate and create some tables previously to users use my app ? ....Do I need to create a specific route like **/populate** and call this every time ?... I'm just looking for some ideas to solve my problem, thanks for your help and time!

**Reply by abdelhai** [reply]
what is the event that triggers the population? like when should it happen?

**Reply by harshitsinghai77** [reply]
Just call the function in `main.py`, I think you would be good. Assuming you need to call this every time the app starts.

**Comment by jnsougata** [comment, marked as answer]
I guess you don't have to make the table at the startup. you can use `put` whenever you want. it will create the table for the user but there's a chance of overwrting the existing data so I will suggest you to use `update` with `set` inside try except block and check for exception... if the key doesn't exist go with `put`
![image](https://user-images.githubusercontent.com/53375272/170807300-c37b10f4-4147-464a-a37a-043b5efd2fba.png)
for example `key` is my target... I use the following to create or update 
```py
    async def add_field(self, key: str, field: Field, force: bool = False) -> Dict[str, Any]:
        """
        adds a field to an existing key.
        if field already exists, old value will be overwritten.
        """
        if not force:
            return await self.update(key=key, updates=[Update.set([field])])
        try:
            return await self.update(key=key, updates=[Update.set([field])])
        except NotFound:
            return await self.put(key=key, field=field)
```

**Reply by Master-Y0da** [reply]
Sorry about late response...but yes ! you are wright!...I just use put when my app calls a router and that's it. Thank you!
